packages/flutter/lib/src/material/text_theme.dart

```diff
@@ -181,50 +181,52 @@ class TextTheme with Diagnosticable {
          (display4 == null && display3 == null && display2 == null && display1 == null && headline == null && title == null &&
           subhead == null && subtitle == null &&
           body2 == null && body1 == null), 'Cannot mix 2014 and 2018 terms in call to TextTheme() constructor.'),
-       headline1 = headline1 ?? display4,
-       headline2 = headline2 ?? display3,
-       headline3 = headline3 ?? display2,
-       headline4 = headline4 ?? display1,
-       headline5 = headline5 ?? headline,
-       headline6 = headline6 ?? title,
-       subtitle1 = subtitle1 ?? subhead,
-       subtitle2 = subtitle2 ?? subtitle,
-       bodyText1 = bodyText1 ?? body2,
-       bodyText2 = bodyText2 ?? body1;
+       // Because of the assert above, the const TextStyle default constructors
```

Comment:
This is the main change in this PR: all the others are just cascading changes from this.